File: sistema_mercado_python/modelos/Venda.py
class Venda:
    def __init__(self, cliente, funcionario):
        self._cliente = cliente
        self._funcionario = funcionario
        self._valor_total = 0.0
        self._cashback = 0.0
       
    def calcular_valor_total(self):
        total = 0.0
        for item in self._cliente.get_carrinho():
            total += item.get_produto().get_preco()

        self._valor_total = total
        self._cashback = total * 0.05 
        self._cliente.adicionar_cashback(self._cashback)

        self._funcionario.adicionar_comissao(total * 0.02)

    def exibir_dados(self):
        print(f"Dados da venda")
        print(f"Cliente: {self._cliente.get_nome()}")
        print(f"Funcionario: {self._funcionario.get_nome()}")
        print(f"Valor total: R${self._valor_total:.2f}")
        print(f"Cashback gerado: R${self._cashback:.2f}")

    def get_cliente(self):
        return self._cliente

    def set_cliente(self, cliente):
        self._cliente = cliente

    def get_funcionario(self):
        return self._funcionario

    def set_funcionario(self, funcionario):
        self._funcionario = funcionario

    def get_valor_total(self):
        return self._valor_total

    def set_valor_total(self, valor_total):
        self._valor_total = valor_total

    def get_cashback(self):
        return self._cashback

    def set_cashback(self, cashback):
        self._cashback = cashback

    def get_data(self):
        return self._data

    def set_data(self, data):
        self._data = data

File: sistema_mercado_java/src/modelo/Venda.java
package modelo;

import java.time.LocalDate;
import java.time.Period;
import java.time.format.DateTimeFormatter;

public class Venda implements Exibivel {

    // Atributos
    private double valor_total;
    private double cashback;
    private Cliente cliente;
    private Funcionario funcionario;
    private LocalDate data;
    private DateTimeFormatter formato = DateTimeFormatter.ofPattern("dd/MM/yyyy");

    // Construtor
    public Venda(Cliente cliente, Funcionario funcionario) {
        this.cliente = cliente;
        this.funcionario = funcionario;

        data = LocalDate.now();

         double total = 0.0;

        // Percorre o carrinho do cliente
        for (ItemVenda item : cliente.getCarrinho()) {
            double preco = item.getProduto().getPreco();
            int quantidade = item.getQuantidade();
            total += preco * quantidade;
        }

       this.valor_total = total;
       this.cashback = total * 0.05;
       cliente.setCashback(this.cashback);
       funcionario.setComissao(this.cashback);
    }

    public String exibirDados() {
        String retorno = "Dados da venda:\n";
        retorno += "Data da venda: " + data.format(formato) + "\n";
        retorno += "Cliente: " + cliente.getNome() + "\n";
        retorno += "Funcionario: " + funcionario.getNome() + "\n";
        retorno += "Valor total: " + valor_total + "\n";
        return retorno;
    }

    // Getters e Setters
    public double getValor_total() {
        return valor_total;
    }

    public void setValor_total(double valor_total) {
        this.valor_total = valor_total;
    }

    public double getCashback() {
        return cashback;
    }

    public void setCashback(double cashback) {
        this.cashback = cashback;
    }

    public Cliente getCliente() {
        return cliente;
    }

    public void setCliente(Cliente cliente) {
        this.cliente = cliente;
    }

    public Funcionario getFuncionario() {
        return funcionario;
    }

    public void setFuncionario(Funcionario funcionario) {
        this.funcionario = funcionario;
    }

}